libs/src/utils/multiProvider.ts

```diff
@@ -9,28 +21,31 @@ const { shuffle } = require('lodash')
  * MultiProvider implements HttpProvider which can be consumed by web3
  * ref for HttpProvider: https://github.com/ChainSafe/web3.js/blob/1.x/packages/web3-providers-http/types/index.d.ts#L46-L66
  */
-class MultiProvider extends Web3.providers.HttpProvider {
+export class MultiProvider extends Web3.providers.HttpProvider {
+  providers: Providers
   /**
    * Creates a MultiProvider
    * @param {Array<string | Provider> | string} - The providers to use.
    */
-  constructor (providers) {
-    let web3Providers = providers
-    if (typeof web3Providers === 'string') {
-      web3Providers = web3Providers.split(',')
-    } else if (!Array.isArray(web3Providers)) {
-      web3Providers = [web3Providers]
+  constructor (providers: Providers | string) {
+    let web3Providers: Providers | StringProviders
+    if (typeof providers === 'string') {
+      web3Providers = providers.split(',') as StringProviders
+    } else if (!Array.isArray(providers)) {
+      web3Providers = [providers]
+    } else {
+      web3Providers = providers
     }
```

Comment:
awesome!!

cc @cheran-senthil too i think you wrote some of this code?

Reply:
yeah was chatting with him about it!